org/postgresql/test/jdbc2/PGPropertyTest.java

```diff
@@ -218,4 +220,26 @@ public void testIsPresentWithParseURLResult() throws Exception
         parsedProperties = Driver.parseURL(TestUtil.getURL() + "&ssl=true" , null);
         Assert.assertTrue("SSL property should be present", PGProperty.SSL.isPresent(parsedProperties));
     }
+    
+    /**
+     * Check whether the isPresent method really works.
+     */
+    public void testPresenceCheck() 
+    {
+        Properties empty = new Properties();
+        Object value = PGProperty.LOG_LEVEL.get(empty);
+        assertNotNull(value);
+        Assert.assertFalse(PGProperty.LOG_LEVEL.isPresent(empty));
+    }
+    
+    public void testNullValue() 
+    {
+        Properties empty = new Properties();
+        assertNull(PGProperty.LOG_LEVEL.getSetString(empty));
+        assertNull(PGProperty.LOG_LEVEL.getSetString(empty));
```

Comment:
We must be able to avoid the NPE in the code, no ?

Dave Cramer

On 2 December 2015 at 11:41, Jeremy Whiting <notifications@github.com>
wrote:

> In org/postgresql/test/jdbc2/PGPropertyTest.java
> <https://github.com/pgjdbc/pgjdbc/pull/438#discussion_r46438994>:
>
> > +    /**
> > +     * Check whether the isPresent method really works.
> > +     */
> > +    public void testPresenceCheck()
> > +    {
> > +        Properties empty = new Properties();
> > +        Object value = PGProperty.LOG_LEVEL.get(empty);
> > +        assertNotNull(value);
> > +        Assert.assertFalse(PGProperty.LOG_LEVEL.isPresent(empty));
> > +    }
> > +
> > +    public void testNullValue()
> > +    {
> > +        Properties empty = new Properties();
> > +        assertNull(PGProperty.LOG_LEVEL.getSetString(empty));
> > +        assertNull(PGProperty.LOG_LEVEL.getSetString(empty));
>
> Well spotted. No I hadn't meant to duplicate line 238. Will remove the
> duplicate.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pgjdbc/pgjdbc/pull/438/files#r46438994>.
>